src/contrib/cluster/Akka.Cluster.Sharding/PersistentShardCoordinator.cs

```diff
@@ -133,6 +134,53 @@ public State Updated(IDomainEvent e)
 
                 return new State(shards ?? Shards, regions ?? Regions, regionProxies ?? RegionProxies, unallocatedShards ?? UnallocatedShards);
             }
+
+            #region Equals
+
+            public override bool Equals(object obj)
+            {
+                var other = obj as State;
+
+                if (ReferenceEquals(other, null)) return false;
+                if (ReferenceEquals(other, this)) return true;
+
+                return Shards.SequenceEqual(other.Shards)
+                    && Regions.Keys.SequenceEqual(other.Regions.Keys)
+                    && RegionProxies.SequenceEqual(other.RegionProxies)
+                    && UnallocatedShards.SequenceEqual(other.UnallocatedShards);
+            }
+
+            public override int GetHashCode()
```

Comment:
The fixed integer values are a smell, and a problem if the same gethashcode is copy pasted everywhere. But other then that its not like we need encryptionlevel hashcodes here.